### PR TITLE
docs: fix links, add references, rename files

### PR DESCRIPTION
**Pull request by PGNetHun**
### Description of the feature or fix

- Renamings: `img` -> `image`, `btn` -> `button`
- Rename files: `img.rst` -> `image.rst`, `butttonmatrix.rst` -> `buttonmatrix.rst`, `disp.rst` -> `display.rst`
- Fix many references/targets and lists
- Remove label text recolor
- Remove accidentally added chinese HTML texts
- Fix many Sphinx domain roles
- Add PNG compression tip to LodePNG decoder document
- Fix places based on PR review feedbacks

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [x] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<module_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

**Comment by kisvegabor** [comment]
Wow, that must be a huge work to fix all these! Thank you! 

> Unfortunately documentation relative links (example: </widgets/buttonmatrix>) still do not work in LVGL documentation site (https://docs.lvgl.io/master/index.html), due to version text can be found in URL (master, 8.3, 8.2, ....), and those links are relative to the root (https://docs.lvgl.io/)

`/widgets/buttonmatrix` worked in v8.3 in the markdown based docs. @kdschlosser do you agree to fix it like [this](https://stackoverflow.com/a/68984203)?

(The current value of [html_baseurl ](https://github.com/lvgl/lvgl/blob/master/docs/conf.py#L133) is incorrect as we don't have the `en/html` part anymore.)

**Comment by kdschlosser** [comment]
I am going to take a look at it now. I just opened #5153 that lists all of the things that doxygen is complaining about. Those things also need to be corrected as well.

**Comment by kdschlosser** [comment]
OH!!! Remember I went over this with you. we need to correct those internal links to use references instead. This is a much cleaner way to handle it. the external links cause a full page reload where the reference only loads the appropriate section of the documentation and update only the one frame. The mechanics of how it works are better.

so what you would use is 

```
:ref:`porting`
```

where the reference is pointing to has to be defined like so.

```
.. _porting:

=======
Porting
=======


.. toctree::
    :maxdepth: 2

    project
    disp
    indev
    tick
    timer_handler
    sleep
    os
    log
    draw

```

You can see the reference definition at the top above the header. when defining a reference it ALWAYS needs to be placed before a section heading and there MUST be one empty line between the reference definition and the heading. When sphinx builds the documentation into the HTML files it will create the internal link using the heading title as the text for the link.

**Comment by kdschlosser** [comment]
so with the widgets/buttonmatrix what you would do is you open up the buttonmatrix.rst file located in docs/widgets. Add to the top of it so it looks like this.

```
.._lv_buttonmatrix:

Button matrix (lv_buttonmatrix)
============================
```

and when you want to internally link to it you would use 

```
:ref:`lv_buttonmatrix`
```

**Comment by kdschlosser** [comment]
Could things I forgot to mention 
You have to keep reference names unique and when defining the reference it MUST always begin with an underscore "_"

I suggest that we come up with some kind of a naming convention for these references. each heading in each of the REsT files should have a reference defined. 

for example in the buttonmatrix.rst file you have these sections.
* Button matrix (lv_buttonmatrix)
* Overview
* Parts and Styles
* Usage
* Button's text
* Control buttons
* One check
* Events
* Keys
* Example
* API

I recommend the "K.I.S.S." (Keep It Stupid Simple) way of handling it.

* Button matrix (lv_buttonmatrix): .._lv_buttonmatrix:
* Overview: .._lv_buttonmatrix_overview:
* Parts and Styles: .._lv_buttonmatrix_parts_and_styles:
* Usage: .._lv_buttonmatrix_usage:
* Button's text: .._lv_buttonmatrix_buttons_text:
* Control buttons: .._lv_buttonmatrix_control_buttons:
* One check: .._lv_buttonmatrix_one_check:
* Events: .._lv_buttonmatrix_events:
* Keys: .._lv_buttonmatrix_keys:
* Example: .._lv_buttonmatrix_example:
* API: .._lv_buttonmatrix_api:

File: docs/libs/qrcode.rst
Comment (kdschlosser) [inline docs/libs/qrcode.rst]: this should be :cpp:func:\`lv_qrcode_create\`

File: docs/libs/qrcode.rst
Comment (kdschlosser) [inline docs/libs/qrcode.rst]: this should be :cpp:func:\`lv_qrcode_update\`

File: docs/libs/qrcode.rst
Comment (kdschlosser) [inline docs/libs/qrcode.rst]: :cpp:func:\`lv_qrcode_set_size\`, :cpp:func:\`lv_qrcode_set_dark_color\`, :cpp:func:\`lv_qrcode_set_light_color\`

File: docs/libs/qrcode.rst
Comment (kdschlosser) [inline docs/libs/qrcode.rst]: :cpp:func:\`lv_qrcode_update\`

File: docs/libs/rlottie.rst
Comment (kdschlosser) [inline docs/libs/rlottie.rst]: Grammar is wrong. I have to look at the entire source file to get the context it which it is being used. I am thinking it should be along the lines of 

> Allows playing Lottie animations in LVGL.

File: docs/libs/rlottie.rst
Comment (kdschlosser) [inline docs/libs/rlottie.rst]: This happened to me a lot when i was going over the files to correct things. 

it should be 
```
- Build the lottie animation to be sized for the intended size 
- it can scale/resize,
```

File: docs/libs/rlottie.rst
Comment (kdschlosser) [inline docs/libs/rlottie.rst:260]: I just want to add a note to this section as a reminder to update it. The ESP-IDF allows for setting memory thresholds so that anything being allocated above a specific set size will get allocated in SPIRAM instead of SRAM.  It does this without having to use the espidf malloc function.

File: docs/libs/tjpgd.rst
Comment (kdschlosser) [inline docs/libs/tjpgd.rst]: Tiny JPEG Decompressor (TJpgDec)

File: docs/others/gridnav.rst
Comment (kdschlosser) [inline docs/others/gridnav.rst]: these an invalid expressions.

This would look better if it read

:cpp:enumerator:\`LV_KEY_NEXT\` or :cpp:enumerator:\`LV_KEY_PREV\`. Optionally you can also use :cpp:func:\`lv_group_focus_next\` or :cpp:func:\`lv_group_focus_prev\`

File: docs/others/imgfont.rst
Comment (kdschlosser) [inline docs/others/imgfont.rst]: Image font (lv_imgfont)

File: docs/others/obj_id.rst
Comment (kdschlosser) [inline docs/others/obj_id.rst]: :cpp:type:\`lv_obj_t\`

too many back ticks

File: docs/others/obj_id.rst
Comment (kdschlosser) [inline docs/others/obj_id.rst]: \`1\`

too many back ticks

File: docs/others/obj_id.rst
Comment (kdschlosser) [inline docs/others/obj_id.rst]: :cpp:expr:\`obj->id\`

additional colon ":" at the end of the expression

File: docs/others/observer.rst
Comment (kdschlosser) [inline docs/others/observer.rst]: you cannot have \<params\> in the extression. It only evaluates actual C/CPP code. So this one should be corrected so that the \<params\> is replaced or remove the :cpp:expr: directive

File: docs/overview/disp.rst
Comment (kdschlosser) [inline docs/overview/disp.rst]: :ref:\`porting\`

File: docs/overview/disp.rst
Comment (kdschlosser) [inline docs/overview/disp.rst]: need to add proper reference

File: docs/overview/disp.rst
Comment (kdschlosser) [inline docs/overview/disp.rst]: need to add proper reference

File: docs/overview/draw.rst
Comment (kdschlosser) [inline docs/overview/draw.rst]: need to add proper reference

File: docs/overview/font.rst
Comment (kdschlosser) [inline docs/overview/font.rst:48]: add additional empty line

File: docs/overview/fs.rst
Comment (kdschlosser) [inline docs/overview/fs.rst]: need to add proper reference

**Comment by kdschlosser** [comment]
Also all of the filenames in the documentation has to be changed so they align with what the header file names are (minus the lv_). so disp.rst needs to get changed to display.rst and anything that points to that file explicitly by name also needs to get changed as well. 

The API section of the documentation relies on the file names being in alignment with the header file names.

**Comment by PGNetHun** [comment]
OK, I will do all the suggested changes.

**Comment by kdschlosser** [comment]
I haven't gone through everything you have changed so far. only made it to about 2/3rds and my eyes started to go sideways. LOL. can only look at that stuff for so long because you start to miss things.

TY very much for doing this. I did a lot when I converted the documentation system to using restructured text instead of markdown. I ran a converter that did the bulk stuff but it got a lot of things wrong and they have to be manually fixed unfortunately. No way around it. I didn't want to make changes and then go back in and make changes again because of updates to LVGL taking place at the same time. With the massive API update it made sense to wait and take care of it all at once.

I can lend a hand with it if you want. I know it is a monster task.  Let me know if you want some help with it.

**Comment by kisvegabor** [comment]
Let's proceed in small steps. 
1. I think once the comments of @kdschlosser are fixed we can merge this PR.
2. @kdschlosser can you write a script to automatically insert `.._lv_buttonmatrix:` and similar tags?  
3. Figure out what to do with the issues listed in  #5153

**Comment by kdschlosser** [comment]
I think I can put together something that can generate those tags for us. Can even have it locate the links and correct them so they use the :ref: directive.

**Comment by kisvegabor** [comment]
That would be amazing!

**Comment by kdschlosser** [comment]
give me a day or so to put it together. The script is not something I am going to do a PR for it will be the changes that get made. I won't submit that PR until all modifications to the documentation get merged in. I don't want to cause a grip of merge conflicts with other PR changes that are getting made to the docs.

**Comment by kisvegabor** [comment]
Ok, I'll merge it when @PGNetHun updates the PR. After that I won't merge any larger docs update until the script is ready and updated the docs.

**Comment by kdschlosser** [comment]
That will work for me....

so hopefully @PGNetHun will see this before going through the effort of manually going in and making those changes.

File: docs/overview/fs.rst
Comment (PGNetHun) [reply docs/overview/fs.rst]: fixed

File: docs/overview/font.rst
Comment (PGNetHun) [reply docs/overview/font.rst:48]: Not sure what you mean

File: docs/overview/draw.rst
Comment (PGNetHun) [reply docs/overview/draw.rst]: fixed

File: docs/overview/disp.rst
Comment (PGNetHun) [reply docs/overview/disp.rst]: fixed

File: docs/overview/disp.rst
Comment (PGNetHun) [reply docs/overview/disp.rst]: fixed

File: docs/overview/disp.rst
Comment (PGNetHun) [reply docs/overview/disp.rst]: fixed

File: docs/others/observer.rst
Comment (PGNetHun) [reply docs/others/observer.rst]: replaced to `params`

File: docs/others/obj_id.rst
Comment (PGNetHun) [reply docs/others/obj_id.rst]: removed

File: docs/others/obj_id.rst
Comment (PGNetHun) [reply docs/others/obj_id.rst]: removed

File: docs/others/obj_id.rst
Comment (PGNetHun) [reply docs/others/obj_id.rst]: removed

File: docs/others/imgfont.rst
Comment (PGNetHun) [reply docs/others/imgfont.rst]: corrected

File: docs/others/gridnav.rst
Comment (PGNetHun) [reply docs/others/gridnav.rst]: corrected

File: docs/libs/tjpgd.rst
Comment (PGNetHun) [reply docs/libs/tjpgd.rst]: updated

File: docs/libs/rlottie.rst
Comment (PGNetHun) [reply docs/libs/rlottie.rst]: fixed

File: docs/libs/rlottie.rst
Comment (PGNetHun) [reply docs/libs/rlottie.rst]: fixed

File: docs/libs/qrcode.rst
Comment (PGNetHun) [reply docs/libs/qrcode.rst]: fixed

File: docs/libs/qrcode.rst
Comment (PGNetHun) [reply docs/libs/qrcode.rst]: fixed

File: docs/libs/qrcode.rst
Comment (PGNetHun) [reply docs/libs/qrcode.rst]: fixed

File: docs/libs/qrcode.rst
Comment (PGNetHun) [reply docs/libs/qrcode.rst]: fixed

File: docs/overview/obj.rst
Comment (kdschlosser) [inline docs/overview/obj.rst]: This is wrong. It should read 

:ref:\`display_multi_display_support\`

The title of the section will automatically get used as the link text.

File: docs/porting/display.rst
Comment (kdschlosser) [inline docs/porting/display.rst]: same thing with this one.

:ref:\`display_features\`

File: docs/widgets/image.rst
Comment (kdschlosser) [inline docs/widgets/image.rst:109]: ref needs to be changed here as well.

File: docs/overview/obj.rst
Comment (PGNetHun) [reply docs/overview/obj.rst]: Well, I thought to keep some labels and only reference other page.
Like the syntax mentioned here:
https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#explicit-targets

File: docs/porting/display.rst
Comment (PGNetHun) [reply docs/porting/display.rst]: oh yes, good catch

File: docs/widgets/image.rst
Comment (PGNetHun) [reply docs/widgets/image.rst:109]: what do you mean?

**Comment by PGNetHun** [comment]
OK, I will not modify more files, this is the first (big) package of doc changes. I will only fix those you find in PR.
I have not checked porting directory, so that will be fixed in the next package (a new PR).

File: docs/widgets/image.rst
Comment (kdschlosser) [reply docs/widgets/image.rst:109]: The same thing as the comment above. 

:ref:\`Animating \<animations\>\` 

needs to be changed to

:ref:\`animations\`

File: docs/overview/obj.rst
Comment (kdschlosser) [reply docs/overview/obj.rst]: There is no need to specify the link text. It will pull the link text in from the heading on the page the reference is defined above. It's just extra keystrokes that you don't need to do unless for some reason you which to change the link text that is being displayed. It's up to up. It's more work is all. 


doing this 

:ref:\`Multi-display support \<display_multi_display_support\>\`

will produce a  the link text

```
Multi-display support
```

and using this

:ref:\`display_multi_display_support\`

will produce the link text
```
Multiple display support
```

Sphinx when using the references will go and see where the reference is which is this

```
.. _display_multi_display_support:

Multiple display support
************************
```

and it grabs that text you see right below the reference definition to use as the text for the link when :ref:\`display_multi_display_support\` is used

**Comment by kdschlosser** [comment]
we can also do the references using this.

https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#automatically-label-sections

This will automatically defines the section references for us. we would need to update the reference pointers to fit the syntax that needs to be used. It is another way it can be done without having to explicitly code in all of the reference definitions.

**Review by kisvegabor — APPROVED** [review]
Okay, let's leave it as it is. It's already a big update. 

I bypass the CI because the failing ESP  build is unrelated. See [here](https://github.com/lvgl/lvgl/pull/5175#issuecomment-1876440100).

@kdschlosser 
It seems pretty useful. We can be sure that at least the "tags" won't be forgotten. 

Regardless to it. Can you write a script which automatically updated the the current links?

**Comment by PGNetHun** [comment]
Thanks!
Most of the links work now.
After @kdschlosser 's script is done, and run, 
I will go through again the pages to check the links and text formattings, and will create another PR (second package).

**Comment by kdschlosser** [comment]
I am going to lay down. I have been up for almost 3 days. I am a little tired. I will work on this in a few hours.

**Comment by kisvegabor** [comment]
Thank you for both of you! 🤗

**Comment by kdschlosser** [comment]
OK i'm back.

**Pull request merged**
